File: ESTRUTURAS/2693/main.py
from sys import stdin


def get_students(q):
    students = []
    for _ in range(q):
        a, s, c = input().split()
        students.append([a, s, int(c)])
    return students


def formatted_print(students):
    for student in sorted(students, key=lambda x: [x[2], x[1], x[0]]):
        print(student[0])


for q in map(int, stdin):
    students = get_students(q)
    formatted_print(students)

File: ESTRUTURAS/2693/main.js
function getStudents(q) {
    let students = [];
    while (q--) {
        let [a, s, c] = lines.shift().split(' ');
        students.push([a, s, parseInt(c)]);
    }
    return students;
}

function formattedPrint(students) {
    students.sort((a, b) => a[2] - b[2] || a[1].localeCompare(b[1]) || a[0].localeCompare(b[0]));
    for (let student of students) {
        console.log(student[0]);
    }
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');
while (q = parseInt(lines.shift())) {
    let students = getStudents(q);
    formattedPrint(students);
}
